cylc/flow/option_parsers.py

```diff
@@ -831,6 +831,9 @@ def cleanup_sysargv(
                     not in ['store_true', 'store_false']
                 ):
                     sys.argv.pop(index)
+            elif arg in [i.split('=')[0] for i in sys.argv]:
+                index = [i.split('=')[0] for i in sys.argv].index(arg)
```

Comment:
Could you compute this list outside of the nested for loop (i.e. at line 823) to avoid repeated recomputation.

Reply:
Yes.